lib/helpers/tasks.rb

```diff
@@ -47,7 +47,12 @@ def run_prefix_command(payload)
 
   def run_command(command)
     message = ''
-    File.open(LOCKFILE, 'w+') do |file|
+    lockfile = if settings.respond_to?(:lockfile=)
+                 settings.lockfile
+               else
+                 '/var/run/puppet_webhook/webhook.lock'
```

Comment:
Is this the best place for a default?

Reply:
It's not used anywhere else and I don't want to force users to have/create a config_file for a config option only used once.

I couldn't think of a better place, it doesn't really belong anywhere else since the configuration option is specific to the method it is in.

Reply:
That's what I mentioned about the config class. I also notice that it's testing for ```lockfile=``` rather than plain ```lockfile.

Reply:
As far as I could tell, you have to check for 'config_option=' or :config_option= due to how Sinatra settings work and how the config_file method loads them into it.

Reply:
Would it be an idea to set (hardcoded) defaults and then load the config?

Reply:
Not sure how without making every config option available globally (something I've been trying to reduce). That and we'd essentially be doing what `Sinatra::ConfigFile` already does and creating another class for parsing config.

This might also be a good time to discuss reducing some configuration options. We can't, nor should we, account for every potential way someone wants to use this.

Reply:
For what it's worth, `settings` just returns the application class itself. https://github.com/sinatra/sinatra/blob/master/lib/sinatra/base.rb#L939

Reply:
One thing that I've done before is to write a routine in the initializer to ensure that the settings always exist, even if they're effectively just noops. I don't know if that's a best practice or not, but it seems to work quite well, and then I don't have to sprinkle my code with conditionals.

https://github.com/puppetlabs/puppet-validator/blob/0462e21773c73682d85ce9398eb89db5a215c194/lib/puppet-validator.rb#L37-L46

Reply:
@binford2k That seems like the best direction forward.